packages/web-frontend/src/sagas.js

```diff
@@ -606,11 +606,17 @@ function* fetchViewData(parameters, errorHandler) {
   const requestResourceUrl = `view?${queryString.stringify(urlParameters)}`;
 
   try {
-    const viewData = yield call(request, requestResourceUrl, errorHandler);
-    return viewData;
+    return yield call(request, requestResourceUrl, errorHandler);
   } catch (error) {
+    let errorMessage = error.message;
+
+    if (error.response) {
+      const json = yield error.response.json();
+      errorMessage = json.error;
+    }
+
     if (error.errorFunction) {
-      yield put(error.errorFunction(error, infoViewKey));
+      yield put(error.errorFunction(errorMessage, infoViewKey));
```

Comment:
Ah this should probably stay as errorFunction(error, infoViewKey) no?

And you removed errorHandler()? I have approved anyway, assuming it works as you want it to

Reply:
yep it is working as we discussed. On line 609 the errorHandler gets passed into the request call and then the request handler returns the correct errorHandler back in the error response. If there is a server error, then a generic errorHandler gets passed back otherwise the specified error handler gets passed. I did a bit of digging and testing and realised that this behaviour is already set up in the saga request handler.

Reply:
> Ah this should probably stay as errorFunction(error, infoViewKey) no?
> 
> And you removed errorHandler()? I have approved anyway, assuming it works as you want it to

No here I dod want errorMessage as this will give me the full error message from the server if there is a user error message present. Thanks for the review, I'm much happier with the code and my understanding than I was before :)